crates/tabby/src/serve/completions.rs

```diff
@@ -172,8 +169,21 @@ fn create_engine(args: &crate::serve::ServeArgs) -> (Box<dyn TextGeneration>, Op
             .expect("Missing authorization field")
             .as_str()
             .expect("Type unmatched");
-        let engine = Box::new(VertexAIEngine::create(api_endpoint, authorization));
-        (engine, Some(VertexAIEngine::prompt_template()))
+
+        if kind == "vertex-ai" {
+            let engine = Box::new(VertexAIEngine::create(api_endpoint, authorization));
+            (engine, Some(VertexAIEngine::prompt_template()))
+        } else if kind == "fastchat" {
+            let model_name = params
```

Comment:
I recommend extract a function `fn get_param(params: Value, key: &str) -> &str` and wrap validation logic inside. Then use `get_param` separately under `vertex-ai` and `fastchat` condition branch.

Reply:
done